packages/xo-web/src/xo-app/home/backed-up-vms.js

```diff
@@ -0,0 +1,239 @@
+      />
+    ),
+    sortCriteria: 'name_description',
+  },
+  {
+    name: _('containersTabName'),
+    itemRenderer: (vm, { pools, hosts }) => {
+      let container
+      return vm.power_state === 'Running' &&
+        (container = hosts[vm.$container]) !== undefined ? (
+        <XoSelect
+          compareContainers={createCompareContainers(vm.$pool)}
+          labelProp='name_label'
+          onChange={host => migrateVm(vm, host)}
+          placeholder={_('homeMigrateTo')}
+          useLongClick
+          value={container}
+          xoType='host'
+        >
+          <Host id={container.id} link />
+        </XoSelect>
+      ) : (
+        (container = pools[vm.$container]) !== undefined && (
+          <Pool id={container.id} link />
+        )
+      )
+    },
+  },
+]
+
+const ACTIONS = [
+  {
+    disabled: vms => some(vms, { power_state: 'Running' }),
+    handler: startVms,
+    icon: 'vm-start',
+    label: _('startVmLabel'),
+  },
+  {
+    disabled: vms => some(vms, { power_state: 'Halted' }),
+    handler: vms => stopVms(vms),
+    icon: 'vm-stop',
+    label: _('stopVmLabel'),
+  },
+  {
+    handler: migrateVms,
+    icon: 'vm-migrate',
+    label: _('migrateVmLabel'),
+  },
+  {
+    handler: snapshotVms,
+    icon: 'vm-snapshot',
+    label: _('snapshotVmLabel'),
+  },
+  {
+    handler: copyVms,
+    icon: 'vm-copy',
+    label: _('copyVmLabel'),
+  },
+  {
+    handler: (vms, { setHomeVmIdsSelection }) => {
+      setHomeVmIdsSelection(map(vms, 'id'))
+    },
+    icon: 'backup',
+    label: _('backupLabel'),
+    redirectOnSuccess: '/backup/new/vms',
+  },
+  {
+    handler: deleteVms,
+    icon: 'vm-delete',
+    label: _('vmRemoveButton'),
+    level: 'danger',
+  },
+]
+
+const GROUPED_ACTIONS = [
+  {
+    disabled: vms => some(vms, _ => _.power_state !== 'Running'),
+    handler: vms => restartVms(vms),
+    icon: 'vm-reboot',
+    label: _('rebootVmLabel'),
+  },
+  {
+    disabled: vms => some(vms, _ => _.power_state !== 'Running'),
+    handler: pauseVms,
+    icon: 'vm-pause',
+    label: _('pauseVmLabel'),
+  },
+  {
+    disabled: vms => some(vms, _ => _.power_state !== 'Running'),
+    handler: suspendVms,
+    icon: 'vm-suspend',
+    label: _('suspendVmLabel'),
+  },
+]
+
+const BackedUpVms = decorate([
+  addSubscriptions({
+    jobs: subscribeBackupNgJobs,
+  }),
+  connectStore(() => ({
+    hosts: createGetObjectsOfType('host'),
+    pools: createGetObjectsOfType('pool'),
+  })),
+  provideState({
+    computed: {
+      backedUpVms: (_, { showBackedUpVms, jobs, vms }) =>
+        uniq(
+          flatMap(jobs, job =>
+            filter(vms, createPredicate(omit(job.vms, 'power_state')))
+          )
+        ),
+      collection: ({ backedUpVms, notBackedUpVms }, { showBackedUpVms }) =>
+        showBackedUpVms ? backedUpVms : notBackedUpVms,
+      notBackedUpVms: ({ backedUpVms }, { showBackedUpVms, vms }) =>
+        showBackedUpVms ? [] : difference(vms, backedUpVms),
```

Comment:
Don't access the property if you don't need it otherwise reaclette will trigger a recompute. Then, you don't need any other optimizations since it's reaclette's job:
```js
collection: (state, { showBackedUpVms }) =>
  showBackedUpVms ? state.backedUpVms : state.notBackedUpVms,
notBackedUpVms: ({ backedUpVms }, { vms }) => difference(vms, backedUpVms),
```